docs/source/development.rst

```diff
@@ -106,14 +106,164 @@ are making:
+
+If you had tried to push the topic branch without using the force flag,
+it would have failed.  This is because non-force pushes only succeed when
+you are only adding new commits to the tip of the existing remote branch.
+When you want to do something other than that, such as insert commits
+in the middle of the branch history (what ``git rebase`` does), or modify a
+commit (what ``git commit --amend`` does) you need to blow away the remote
+version of your branch and replace it with the local version.  This is
+exactly what a force push does.
+
+**IMPORTANT**: Never use the force flag to push to ``upstream``.  Never use
+the force flag to push to ``develop`` or ``master``.  Only use the force 
+flag on your repository and on your topic branches.  Otherwise you run the 
+risk of screwing up the mainline branches, which will require manual 
+intervention by a senior developer and manual changes by every downstream 
+developer.  That is a recoverable situation, but also one that we would
+rather avoid.
+(Note: a hint that this has happened is that one of the above listed
+merge commands that uses the ``--ff-only`` flag to merge a remote mainline
+branch into a local mainline branch fails.)
+
+One task that you might be asked to do before your topic branch can be
+merged is "squashing your commits."  We want the git history to be clean
+and informative, and we do that by crafting one and only one commit
+message per logical change.  In the normal course of development (unless
+one is constantly committing with the ``--amend`` flag) many intermediate
+commits can be created that should be squashed down to (usually) one before
+it can be merged.  Do this with (assuming you are in your topic branch):
+
+.. code-block:: bash
+
+    $ git rebase -i develop
+
+This will launch an editor that will give you control over how you want
+to structure your commits.  Usually you just want to "pick" the first
+commit and "squash" all of the subsequent commits, and then ensure that
+the final commit message is clean (best practice is to give a short
+summary of the change on the first line, a blank line, and then a more
+detailed description of the change following, with the issue number
+-- if there is one -- in the detailed description).  More information
+about the interactive rebase process can be found 
+`here <https://help.github.com/articles/about-git-rebase/>`_.
+Once the commits are to your liking, you can push the branch to your
+remote repository (which will require a force push if you reordered
+or deleted commits that existed in the remote version of the branch).
+
+Once your pull request has been merged into ``develop``, you can close
+the pull request and delete the remote branch in the GitHub interface.
+Locally, run this command to delete the topic branch:
+
+.. code-block:: bash
+
+    $ git branch -D TOPIC_BRANCH_NAME
+
+Customizing git's `configuration <http://git-scm.com/book/en/v2/Customizing-Git-Git-Configuration>`_ 
```

Comment:
I think we might be veering off topic a bit here with the last three paragraphs. I would definitely delete the bit about aliases --- you never know how these might interact with someone's environment.